Create bibliotheca-hertziana-max-planck-institute-for-art-history.csl

hawkenbury opened the pull request:
Initial Commit : Bibliotheca Hertziana Max Planck Institute for Art History

hawkenbury reviewed (commented):
Added full-stop to initials in authors macro

hawkenbury commented:
Hi, 
Do I need to do anything else for this to be published?

Best Wishes
Chris

adam3smith commented on bibliotheca-hertziana-max-planck-institute-for-art-history.csl:
```diff
@@ -0,0 +1,317 @@
+            <text variable="title" font-style="italic" suffix=", "/>
+          </group>
+          <group delimiter="; ">
+            <date variable="original-date" prefix=" (" suffix="), ">
```
you'd get this twice for books since it's also in the date-reference macro

hawkenbury replied:
Date reference is used once in th ecitation and once in th ebibliography

adam3smith commented on bibliotheca-hertziana-max-planck-institute-for-art-history.csl:
```diff
@@ -0,0 +1,317 @@
+          <else-if type="article-journal" match="any">
+            <group delimiter=" ">
+              <group delimiter=", ">
+                <group delimiter=": ">
```
remove groups with just one element -- they have no effect and make the style harder to read

hawkenbury replied:
removed

adam3smith commented on bibliotheca-hertziana-max-planck-institute-for-art-history.csl:
```diff
@@ -0,0 +1,317 @@
+            </group>
+            <date date-parts="year" form="text" variable="issued" suffix="."/>
+            <group delimiter=" ">
+              <label strip-periods="true" suffix=". " variable="page" form="short"/>
```
please clean up the strip-periods -- suffix="." combinations throughout -- you're removing periods and putting them back in.

hawkenbury replied:
removed

adam3smith commented on bibliotheca-hertziana-max-planck-institute-for-art-history.csl:
```diff
@@ -0,0 +1,317 @@
+            <group delimiter=" ">
+              <group delimiter=", ">
+                <group>
+                  <text value="Manuscript"/>
```
you can use a term for this; all item types are available as terms

adam3smith commented on bibliotheca-hertziana-max-planck-institute-for-art-history.csl:
```diff
@@ -0,0 +1,317 @@
+          </else-if>
+          <else>
+            <group delimiter=" ">
+              <text value="This reference type is not currently supported"/>
```
we'd much rather avoid this. CSL is written so as to allow sensible fallbacks for item types not covered by a citation style.

hawkenbury replied:
removed

adam3smith reviewed (changes requested):
Sorry for the delay -- some questions, some things you'd need to fix.

hawkenbury commented on bibliotheca-hertziana-max-planck-institute-for-art-history.csl:
```diff
@@ -0,0 +1,317 @@
+      <term name="close-quote">”</term>
+      <term name="open-inner-quote">‘</term>
+      <term name="close-inner-quote">’</term>
+      <term name="page-range-delimiter">&#8211;</term>
```
removed

hawkenbury commented on bibliotheca-hertziana-max-planck-institute-for-art-history.csl:
```diff
@@ -0,0 +1,317 @@
+    <names variable="author">
+      <name form="short" delimiter=" / " delimiter-precedes-et-al="never" delimiter-precedes-last="always" et-al-min="4" et-al-use-first="1"/>
+      <substitute>
+        <group>
```
this is the style that the institution wants - the rules are complicated but this is correct

hawkenbury commented on bibliotheca-hertziana-max-planck-institute-for-art-history.csl:
```diff
@@ -0,0 +1,317 @@
+      <group delimiter=":">
+        <group delimiter=" ">
+          <text macro="contributors-short"/>
+          <group delimiter="; ">
```
removed

hawkenbury commented:
Thanks for looking at that
I have updated things in line with your comments

Best Wishes

---------------------------------------
Chris Tomlinson
Software Developer
Bibliotheca Hertziana



From: Sebastian Karcher ***@***.***>
Sent: 21 November 2023 06:05
To: citation-style-language/styles ***@***.***>
Cc: Tomlinson, Christopher ***@***.***>; Author ***@***.***>
Subject: Re: [citation-style-language/styles] Create bibliotheca-hertziana-max-planck-institute-for-art-history.csl (PR #6525)


@adam3smith requested changes on this pull request.

Sorry for the delay -- some questions, some things you'd need to fix.

________________________________

In bibliotheca-hertziana-max-planck-institute-for-art-history.csl<https://github.com/citation-style-language/styles/pull/6525#discussion_r1400028450>:

> +    <link href="http://www.zotero.org/styles/bmj" rel="template"/>

+    <link href="https://www.biblhertz.it/3170244/rjbh_editorial_guidelines_en.pdf" rel="documentation"/>

+    <author>

+      <name>Chris Tomlinson</name>

+    </author>

+    <category citation-format="author-date"/>

+    <updated>2023-11-20T12:32:44+00:00</updated>

+    <rights license="http://creativecommons.org/licenses/by-sa/3.0/">This work is licensed under a Creative Commons Attribution-ShareAlike 3.0 License</rights>

+  </info>

+  <locale xml:lang="en">

+    <terms>

+      <term name="open-quote">“</term>

+      <term name="close-quote">”</term>

+      <term name="open-inner-quote">‘</term>

+      <term name="close-inner-quote">’</term>

+      <term name="page-range-delimiter">&#8211;</term>
⬇️ Suggested change

-      <term name="page-range-delimiter">&#8211;</term>

this is already set to an en-dash

________________________________

In bibliotheca-hertziana-max-planck-institute-for-art-history.csl<https://github.com/citation-style-language/styles/pull/6525#discussion_r1400029057>:

> +      </if>

+      <else>

+        <text term="no date" form="short"/>

+      </else>

+    </choose>

+  </macro>

+  <citation disambiguate-add-year-suffix="true" disambiguate-add-names="true" disambiguate-add-givenname="true">

+    <sort>

+      <key variable="issued" sort="ascending"/>

+      <key macro="author"/>

+    </sort>

+    <layout delimiter=", ">

+      <group delimiter=":">

+        <group delimiter=" ">

+          <text macro="contributors-short"/>

+          <group delimiter="; ">

remove this group (it doesn't actually do anything). Also, you really want "dimensions" there are as a variable?

________________________________

In bibliotheca-hertziana-max-planck-institute-for-art-history.csl<https://github.com/citation-style-language/styles/pull/6525#discussion_r1400029625>:

> +  </macro>

+  <macro name="point-locators">

+    <group>

+      <choose>

+        <if locator="page" match="none">

+          <label variable="locator" form="short" suffix=" "/>

+        </if>

+      </choose>

+      <text variable="locator"/>

+    </group>

+  </macro>

+  <macro name="contributors-short">

+    <names variable="author">

+      <name form="short" delimiter=" / " delimiter-precedes-et-al="never" delimiter-precedes-last="always" et-al-min="4" et-al-use-first="1"/>

+      <substitute>

+        <group>

You're sure you don't want editor in here? It's very unusual to cite an edited volume by its title rather than by its editors.
Ditto for the author macro above.

________________________________

In bibliotheca-hertziana-max-planck-institute-for-art-history.csl<https://github.com/citation-style-language/styles/pull/6525#discussion_r1400030181>:

> +            </date>

+          </group>

+        </group>

+      </if>

+    </choose>

+    <text variable="DOI" text-case="lowercase" prefix=" DOI: https://doi.org/"/>

+  </macro>

+  <macro name="title">

+    <group>

+      <choose>

+        <if type="book" match="any">

+          <group>

+            <text variable="title" font-style="italic" suffix=", "/>

+          </group>

+          <group delimiter="; ">

+            <date variable="original-date" prefix=" (" suffix="), ">

you'd get this twice for books since it's also in the date-reference macro

________________________________

In bibliotheca-hertziana-max-planck-institute-for-art-history.csl<https://github.com/citation-style-language/styles/pull/6525#discussion_r1400030614>:

> +              <label strip-periods="true" suffix="." variable="volume" form="short"/>

+              <text variable="volume" suffix=","/>

+            </group>

+            <date date-parts="year" form="text" variable="issued" suffix="."/>

+            <group delimiter=" ">

+              <label strip-periods="true" suffix=". " variable="page" form="short"/>

+              <text variable="page"/>

+            </group>

+            <group>

+              <text variable="URL" prefix="URL: " suffix="."/>

+            </group>

+          </if>

+          <else-if type="article-journal" match="any">

+            <group delimiter=" ">

+              <group delimiter=", ">

+                <group delimiter=": ">

remove groups with just one element -- they have no effect and make the style harder to read

________________________________

In bibliotheca-hertziana-max-planck-institute-for-art-history.csl<https://github.com/citation-style-language/styles/pull/6525#discussion_r1400030964>:

> +              <text variable="number-of-volumes" form="short" suffix=" vols.,"/>

+            </group>

+            <group delimiter=" ">

+              <text variable="collection-title" suffix=","/>

+              <text variable="collection-number" suffix=","/>

+              <text macro="edition" prefix=" " suffix=", "/>

+              <text macro="publisher" prefix=" " suffix=","/>

+              <text variable="publisher-place" suffix=","/>

+            </group>

+            <group delimiter=" ">

+              <label strip-periods="true" suffix="." variable="volume" form="short"/>

+              <text variable="volume" suffix=","/>

+            </group>

+            <date date-parts="year" form="text" variable="issued" suffix="."/>

+            <group delimiter=" ">

+              <label strip-periods="true" suffix=". " variable="page" form="short"/>

please clean up the strip-periods -- suffix="." combinations throughout -- you're removing periods and putting them back in.

________________________________

In bibliotheca-hertziana-max-planck-institute-for-art-history.csl<https://github.com/citation-style-language/styles/pull/6525#discussion_r1400031496>:

> +                <group>

+                  <text value="Ph.D Thesis"/>

+                </group>

+                <text macro="publisher" prefix=" "/>

+              </group>

+              <date date-parts="year" form="text" variable="issued" suffix="."/>

+              <group>

+                <text variable="URL" prefix=" URL: " suffix="."/>

+              </group>

+            </group>

+          </else-if>

+          <else-if type="manuscript" match="any">

+            <group delimiter=" ">

+              <group delimiter=", ">

+                <group>

+                  <text value="Manuscript"/>

you can use a term for this; all item types are available as terms

________________________________

In bibliotheca-hertziana-max-planck-institute-for-art-history.csl<https://github.com/citation-style-language/styles/pull/6525#discussion_r1400032007>:

> +            <group delimiter=" ">

+              <group delimiter=", ">

+                <group>

+                  <text value="Manuscript"/>

+                </group>

+                <text macro="publisher" prefix=" "/>

+              </group>

+              <date date-parts="year" form="text" variable="issued" suffix="."/>

+              <group>

+                <text variable="URL" prefix=" URL: " suffix="."/>

+              </group>

+            </group>

+          </else-if>

+          <else>

+            <group delimiter=" ">

+              <text value="This reference type is not currently supported"/>

we'd much rather avoid this. CSL is written so as to allow sensible fallbacks for item types not covered by a citation style.

—
Reply to this email directly, view it on GitHub<https://github.com/citation-style-language/styles/pull/6525#pullrequestreview-1741118550>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/ABGB7SFBKROVSV6FL5LFB2TYFQZAHAVCNFSM6AAAAAAXVOVNZGVHI2DSMVQWIX3LMV43YUDVNRWFEZLROVSXG5CSMV3GSZLXHMYTONBRGEYTQNJVGA>.
You are receiving this because you authored the thread.Message ID: ***@***.***>

adam3smith commented:
OK, last thing -- any reason this is an in-text style? As I understand the style guide, the citations are placed into footnotes, so the style should be `class="note"`

hawkenbury commented:
Ok – I have updated that

From: Sebastian Karcher ***@***.***>
Sent: 28 November 2023 05:05
To: citation-style-language/styles ***@***.***>
Cc: Tomlinson, Christopher ***@***.***>; Author ***@***.***>
Subject: Re: [citation-style-language/styles] Create bibliotheca-hertziana-max-planck-institute-for-art-history.csl (PR #6525)


OK, last thing -- any reason this is an in-text style? As I understand the style guide, the citations are placed into footnotes, so the style should be class="note"

—
Reply to this email directly, view it on GitHub<https://github.com/citation-style-language/styles/pull/6525#issuecomment-1829036806>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/ABGB7SEJ2GXVQTZD3IBJGK3YGVPFJAVCNFSM6AAAAAAXVOVNZGVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMYTQMRZGAZTMOBQGY>.
You are receiving this because you authored the thread.Message ID: ***@***.******@***.***>>

hawkenbury commented:
Hi everyone,

I have updated things to comply with your comments
Please could you publish the style.

Best Wishes


---------------------------------------
Chris Tomlinson
Software Developer
Bibliotheca Hertziana



From: Tomlinson, Christopher
Sent: 22 November 2023 17:07
To: 'citation-style-language/styles' ***@***.***>
Subject: RE: [citation-style-language/styles] Create bibliotheca-hertziana-max-planck-institute-for-art-history.csl (PR #6525)

Thanks for looking at that
I have updated things in line with your comments

Best Wishes

---------------------------------------
Chris Tomlinson
Software Developer
Bibliotheca Hertziana



From: Sebastian Karcher ***@***.***>
Sent: 21 November 2023 06:05
To: citation-style-language/styles ***@***.***>
Cc: Tomlinson, Christopher ***@***.***>; Author ***@***.***>
Subject: Re: [citation-style-language/styles] Create bibliotheca-hertziana-max-planck-institute-for-art-history.csl (PR #6525)


@adam3smith requested changes on this pull request.

Sorry for the delay -- some questions, some things you'd need to fix.

________________________________

In bibliotheca-hertziana-max-planck-institute-for-art-history.csl<https://github.com/citation-style-language/styles/pull/6525#discussion_r1400028450>:

> +    <link href="http://www.zotero.org/styles/bmj" rel="template"/>

+    <link href="https://www.biblhertz.it/3170244/rjbh_editorial_guidelines_en.pdf" rel="documentation"/>

+    <author>

+      <name>Chris Tomlinson</name>

+    </author>

+    <category citation-format="author-date"/>

+    <updated>2023-11-20T12:32:44+00:00</updated>

+    <rights license="http://creativecommons.org/licenses/by-sa/3.0/">This work is licensed under a Creative Commons Attribution-ShareAlike 3.0 License</rights>

+  </info>

+  <locale xml:lang="en">

+    <terms>

+      <term name="open-quote">“</term>

+      <term name="close-quote">”</term>

+      <term name="open-inner-quote">‘</term>

+      <term name="close-inner-quote">’</term>

+      <term name="page-range-delimiter">&#8211;</term>
⬇️ Suggested change

-      <term name="page-range-delimiter">&#8211;</term>

this is already set to an en-dash

________________________________

In bibliotheca-hertziana-max-planck-institute-for-art-history.csl<https://github.com/citation-style-language/styles/pull/6525#discussion_r1400029057>:

> +      </if>

+      <else>

+        <text term="no date" form="short"/>

+      </else>

+    </choose>

+  </macro>

+  <citation disambiguate-add-year-suffix="true" disambiguate-add-names="true" disambiguate-add-givenname="true">

+    <sort>

+      <key variable="issued" sort="ascending"/>

+      <key macro="author"/>

+    </sort>

+    <layout delimiter=", ">

+      <group delimiter=":">

+        <group delimiter=" ">

+          <text macro="contributors-short"/>

+          <group delimiter="; ">

remove this group (it doesn't actually do anything). Also, you really want "dimensions" there are as a variable?

________________________________

In bibliotheca-hertziana-max-planck-institute-for-art-history.csl<https://github.com/citation-style-language/styles/pull/6525#discussion_r1400029625>:

> +  </macro>

+  <macro name="point-locators">

+    <group>

+      <choose>

+        <if locator="page" match="none">

+          <label variable="locator" form="short" suffix=" "/>

+        </if>

+      </choose>

+      <text variable="locator"/>

+    </group>

+  </macro>

+  <macro name="contributors-short">

+    <names variable="author">

+      <name form="short" delimiter=" / " delimiter-precedes-et-al="never" delimiter-precedes-last="always" et-al-min="4" et-al-use-first="1"/>

+      <substitute>

+        <group>

You're sure you don't want editor in here? It's very unusual to cite an edited volume by its title rather than by its editors.
Ditto for the author macro above.

________________________________

In bibliotheca-hertziana-max-planck-institute-for-art-history.csl<https://github.com/citation-style-language/styles/pull/6525#discussion_r1400030181>:

> +            </date>

+          </group>

+        </group>

+      </if>

+    </choose>

+    <text variable="DOI" text-case="lowercase" prefix=" DOI: https://doi.org/"/>

+  </macro>

+  <macro name="title">

+    <group>

+      <choose>

+        <if type="book" match="any">

+          <group>

+            <text variable="title" font-style="italic" suffix=", "/>

+          </group>

+          <group delimiter="; ">

+            <date variable="original-date" prefix=" (" suffix="), ">

you'd get this twice for books since it's also in the date-reference macro

________________________________

In bibliotheca-hertziana-max-planck-institute-for-art-history.csl<https://github.com/citation-style-language/styles/pull/6525#discussion_r1400030614>:

> +              <label strip-periods="true" suffix="." variable="volume" form="short"/>

+              <text variable="volume" suffix=","/>

+            </group>

+            <date date-parts="year" form="text" variable="issued" suffix="."/>

+            <group delimiter=" ">

+              <label strip-periods="true" suffix=". " variable="page" form="short"/>

+              <text variable="page"/>

+            </group>

+            <group>

+              <text variable="URL" prefix="URL: " suffix="."/>

+            </group>

+          </if>

+          <else-if type="article-journal" match="any">

+            <group delimiter=" ">

+              <group delimiter=", ">

+                <group delimiter=": ">

remove groups with just one element -- they have no effect and make the style harder to read

________________________________

In bibliotheca-hertziana-max-planck-institute-for-art-history.csl<https://github.com/citation-style-language/styles/pull/6525#discussion_r1400030964>:

> +              <text variable="number-of-volumes" form="short" suffix=" vols.,"/>

+            </group>

+            <group delimiter=" ">

+              <text variable="collection-title" suffix=","/>

+              <text variable="collection-number" suffix=","/>

+              <text macro="edition" prefix=" " suffix=", "/>

+              <text macro="publisher" prefix=" " suffix=","/>

+              <text variable="publisher-place" suffix=","/>

+            </group>

+            <group delimiter=" ">

+              <label strip-periods="true" suffix="." variable="volume" form="short"/>

+              <text variable="volume" suffix=","/>

+            </group>

+            <date date-parts="year" form="text" variable="issued" suffix="."/>

+            <group delimiter=" ">

+              <label strip-periods="true" suffix=". " variable="page" form="short"/>

please clean up the strip-periods -- suffix="." combinations throughout -- you're removing periods and putting them back in.

________________________________

In bibliotheca-hertziana-max-planck-institute-for-art-history.csl<https://github.com/citation-style-language/styles/pull/6525#discussion_r1400031496>:

> +                <group>

+                  <text value="Ph.D Thesis"/>

+                </group>

+                <text macro="publisher" prefix=" "/>

+              </group>

+              <date date-parts="year" form="text" variable="issued" suffix="."/>

+              <group>

+                <text variable="URL" prefix=" URL: " suffix="."/>

+              </group>

+            </group>

+          </else-if>

+          <else-if type="manuscript" match="any">

+            <group delimiter=" ">

+              <group delimiter=", ">

+                <group>

+                  <text value="Manuscript"/>

you can use a term for this; all item types are available as terms

________________________________

In bibliotheca-hertziana-max-planck-institute-for-art-history.csl<https://github.com/citation-style-language/styles/pull/6525#discussion_r1400032007>:

> +            <group delimiter=" ">

+              <group delimiter=", ">

+                <group>

+                  <text value="Manuscript"/>

+                </group>

+                <text macro="publisher" prefix=" "/>

+              </group>

+              <date date-parts="year" form="text" variable="issued" suffix="."/>

+              <group>

+                <text variable="URL" prefix=" URL: " suffix="."/>

+              </group>

+            </group>

+          </else-if>

+          <else>

+            <group delimiter=" ">

+              <text value="This reference type is not currently supported"/>

we'd much rather avoid this. CSL is written so as to allow sensible fallbacks for item types not covered by a citation style.

—
Reply to this email directly, view it on GitHub<https://github.com/citation-style-language/styles/pull/6525#pullrequestreview-1741118550>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/ABGB7SFBKROVSV6FL5LFB2TYFQZAHAVCNFSM6AAAAAAXVOVNZGVHI2DSMVQWIX3LMV43YUDVNRWFEZLROVSXG5CSMV3GSZLXHMYTONBRGEYTQNJVGA>.
You are receiving this because you authored the thread.Message ID: ***@***.***>

adam3smith commented:
Thanks & sorry for the delay

hawkenbury commented:
Many thanks Sebastian

From: Sebastian Karcher ***@***.***>
Sent: 17 January 2024 04:18
To: citation-style-language/styles ***@***.***>
Cc: Tomlinson, Christopher ***@***.***>; Author ***@***.***>
Subject: Re: [citation-style-language/styles] Create bibliotheca-hertziana-max-planck-institute-for-art-history.csl (PR #6525)


Thanks & sorry for the delay

—
Reply to this email directly, view it on GitHub<https://github.com/citation-style-language/styles/pull/6525#issuecomment-1894871275>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/ABGB7SCYDF6B3RKMFZ7HFVLYO47FJAVCNFSM6AAAAAAXVOVNZGVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMYTQOJUHA3TCMRXGU>.
You are receiving this because you authored the thread.Message ID: ***@***.***>

hawkenbury commented:
Hi Sebastian,

Many thanks for all of your help.
I was just wondering about a problem with the Zotero api with relation to our style below;

This works (using raw.github to load the style);

https://api.zotero.org/groups/4999676/items?tag=hsah:0107&format=bib&style=https://raw.githubusercontent.com/citation-style-language/styles/master/bibliotheca-hertziana-max-planck-institute-for-art-history.csl&linkwrap=1

But this doesn’t work (using the style name as mentioned in the Zotero api);
https://api.zotero.org/groups/4999676/items?tag=hsah:0107&locale=en-GB&style=bibliotheca-hertziana-max-planck-institute-for-art-history&linkwrap=1&format=bib

This does work with the style name;
https://api.zotero.org/groups/4999676/items?tag=hsah:0107&locale=en-GB&style=bibliotheca-hertziana-max-planck-institute-for-art-history&linkwrap=1&format=json

Am I doing something wrong here?

Best Wishes


---------------------------------------
Chris Tomlinson
Software Developer
Bibliotheca Hertziana









From: Tomlinson, Christopher
Sent: 17 January 2024 09:51
To: 'citation-style-language/styles' ***@***.***>
Subject: RE: [citation-style-language/styles] Create bibliotheca-hertziana-max-planck-institute-for-art-history.csl (PR #6525)

Many thanks Sebastian

From: Sebastian Karcher ***@***.******@***.***>>
Sent: 17 January 2024 04:18
To: citation-style-language/styles ***@***.******@***.***>>
Cc: Tomlinson, Christopher ***@***.******@***.***>>; Author ***@***.******@***.***>>
Subject: Re: [citation-style-language/styles] Create bibliotheca-hertziana-max-planck-institute-for-art-history.csl (PR #6525)


Thanks & sorry for the delay

—
Reply to this email directly, view it on GitHub<https://github.com/citation-style-language/styles/pull/6525#issuecomment-1894871275>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/ABGB7SCYDF6B3RKMFZ7HFVLYO47FJAVCNFSM6AAAAAAXVOVNZGVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMYTQOJUHA3TCMRXGU>.
You are receiving this because you authored the thread.Message ID: ***@***.******@***.***>>
